invenio_records_rest/ext.py

```
@@ -44,10 +101,25 @@ def init_app(self, app):
         app.register_blueprint(
             create_blueprint(app.config["RECORDS_REST_ENDPOINTS"])
         )
-        app.extensions['invenio-records-rest'] = self
+        app.extensions['invenio-records-rest'] = _RecordRESTState(app)
 
     def init_config(self, app):
         """Initialize configuration."""
+        app.config.setdefault(
+            "RECORDS_REST_DEFAULT_READ_PERMISSION_FACTORY",
+            "invenio_records.permissions:read_permission_factory")
+        app.config.setdefault(
+            "RECORDS_REST_DEFAULT_CREATE_PERMISSION_FACTORY",
+            "invenio_records.permissions:create_permission_factory")
+        app.config.setdefault(
+            "RECORDS_REST_DEFAULT_UPDATE_PERMISSION_FACTORY",
+            "invenio_records.permissions:update_permission_factory")
+        app.config.setdefault(
+            "RECORDS_REST_DEFAULT_DELETE_PERMISSION_FACTORY",
+            "invenio_records.permissions:delete_permission_factory")
+
+        app.config.setdefault("RECORDS_REST_LOGIN_ENDPOINT", "security.login")
```

Do we redirect in the rest API? shouldn't we just return a 401 error.